### ISSUE TITLE
Canopy liquid and snow capacities not enforced consistently?

### ISSUE THREAD

**Issue by billsacks** [ESCOMP/ctsm#699]
I'm not sure if this is actually a bug, but it looks weird to me, and I'd like to at least get someone to sign off on this:

In CanopyHydrology, there is this code for enforcing canopy liquid and snow capacities (corresponding to equations 2.7.8 and 2.7.9 in https://escomp.github.io/ctsm-docs/doc/build/html/tech_note/Hydrology/CLM50_Tech_Note_Hydrology.html):

https://github.com/ESCOMP/ctsm/blob/89ff666b6d73011c966df52fbfc1d203b1287a2c/src/biogeophys/CanopyHydrologyMod.F90#L318-L330

What I don't understand is why there is a conditional on temperature. It looks to me like, if it's above freezing, snow is allowed to accumulate beyond the specified capacity, and if it's below freezing, liquid is allowed to accumulate beyond the specified capacity. Maybe that's dealt with elsewhere (and in any case, the problem probably isn't too large, or at least probably doesn't persist for two many time steps) but this seems weird to me: I would have expected both snow and liquid capacities to be enforced regardless of the temperature. It seems like the tech note makes no mention of temperatures with respect to enforcing the canopy liquid and snow capacities.

@olyson @swensosc @JustinPerket are any of you able to comment on this?

**Comment by olyson** [comment]
I don't really understand the need for this either.  There's no paper on this and Justin is unlikely to respond. Seems like it could be removed and tested to see what effect it has.  I'm happy to do the testing if this is the way we want to go.

**Comment by billsacks** [comment]
Thanks, @olyson . I just talked with @swensosc and it sounds like he also thinks this conditional shouldn't be here - i.e., this block of code should compute both excess liquid and excess snow runoff, regardless of temperature. @olyson if you'd be willing to look at the scientific impact, that would be great. I haven't talked with Dave about it yet, so I'll leave it to you whether you want to check with him first.

**Comment by olyson** [comment]
I'll just test it with an I2000 case and see what happens.

**Comment by olyson** [comment]
Dave suggested creating a tag with this change and folding it into the CTSM testing that I'll be doing, if you think that is acceptable.
On the subject of CTSM testing, Dave and I talked and thought that we would re-run a section near the end of a control historical simulation with the tag in question and examine our regular diagnostics. 
So, our first comparison would be the tag with this change compared to the initial CLM release tag.  This would thus include ctsm1.0.dev001 in our test.  If we saw larger differences than hoped for, we would then back up to the next major tag you've identified.
Let me know what you think of this.
